components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,193 @@
+import pg from "pg";
+
 export default {
   type: "app",
   app: "postgresql",
-  propDefinitions: {},
+  propDefinitions: {
+    table: {
+      type: "string",
+      label: "Table",
+      description: "Database table",
+      async options() {
+        return this.getTables();
+      },
+    },
+    column: {
+      type: "string",
+      label: "Column",
+      description: "The name of a column in the table to use for deduplication. Defaults to the table's primary key",
+      async options({ table }) {
+        return this.getColumns(table);
+      },
+    },
+    query: {
+      type: "string",
+      label: "SQL Query",
+      description: "Your custom SQL query",
+    },
+    value: {
+      type: "string",
+      label: "Lookup Value",
+      description: "Value to search for",
+      async options({
+        table, column,
+      }) {
+        return this.getColumnValues(table, column);
+      },
+    },
+    rowValues: {
+      type: "object",
+      label: "Row Values",
+      description: "Enter the column names and respective values as key/value pairs",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async getClient() {
+      const { Client } = pg;
+      const {
+        user,
+        password,
+        host,
+        port,
+        database,
+      } = this.$auth;
+      const connectionString = `postgresql://${user}:${password}@${host}:${port}/${database}`;
+      const client = new Client({
+        connectionString,
+      });
+      await client
+        .connect()
+        .catch((err) => console.error("Connection error", err.stack));
+      return client;
+    },
+    async endClient(client) {
+      client.end();
+    },
+    /**
+     * Executes SQL query and returns the resulting rows
+     * @param query {string} SQL query to execute
+     * @returns Array of rows returned from the given SQL query
+     */
+    async executeQuery(query) {
+      const client = await this.getClient();
+      const { rows } = await client.query(query);
+      await this.endClient(client);
+      return rows;
```

Comment:
I think you might consider getting errors running the query and if that happens you are not ending the connection so this is my suggestion:
```suggestion
      try {
        const { rows } = await client.query(query);
        return rows;
      } finally {
        await this.endClient(client);
      }
```